### DISCUSSION TITLE
Idea for a multi-tenancy app

### DISCUSSION THREAD

**Discussion by Falci** [supabase/supabase#21451] [Questions]
Hey there, I want to build an app that will be multi-tenant.
Also, I don't want to use email for users account. I would prefer just username/password. Or even better: tenant/username/password.

The super admin (me) can use email/password.

Here's my plan: in the admin page, I can create new tenants and users for them.
I can just join the `tenant_id` and the username and create a fake email: 

```js
const email = `user+${tenant_id}+${username}@example.com` ;
supa.auth.signUp({ email, password });
```

If I disable email validation, it should be ok, right?
First, I'd like to know what do you think about this strategy.

Also, I have another question about updating password:
Assuming that I have some sort of ACL, how can a privileged tenant user change the password of another user of the same tenant?
Is there a way to use `supabase.auth.` functions for that? Or do I need my own procedures ?

**Comment by GaryAustin1** [comment]
I've seen other users discuss going down the fake email path.  I've not seen anything to point you to of a guide or more finished solution.

The only built in option to update a user's password without logging in is with a service_role admin auth updateUser call. 

The other approaches to multi-tenant are using app metadata and the user jwt or a tenant table with the users id mapping to there role.  Either can then be used in RLS.
